packages/web-components/src/menu/menu.ts

```diff
@@ -0,0 +1,204 @@
+      if (Array.isArray(prev) && prev.length) {
+        this.cleanupTriggerEventListeners(prev[0]);
+      }
+
+      if (next && next.length) {
+        this._trigger = next[0];
+
+        this._trigger.setAttribute('aria-haspopup', 'true');
+        this._trigger.setAttribute('aria-expanded', `${this.expanded}`);
+
+        // Add new event listeners
+        if (this.openOnHover) {
+          this._trigger.addEventListener('mouseover', this.toggleExpanded);
+        } else {
+          this._trigger.addEventListener('click', this.toggleExpanded);
+        }
+      }
+    }
+  }
+
+  // Declare expanded as an observable boolean property
+  @observable
+  @attr({ mode: 'boolean' })
+  public expanded: boolean = false;
+
+  // Method to update the expanded property when it changes
+  protected expandedChanged(): void {
+    if (this.$fastController.isConnected && this._trigger) {
+      this._trigger.setAttribute('aria-expanded', `${this.expanded}`);
+    }
+    Updates.enqueue(() => {
+      this.setPositioning();
+      this.focus();
+    });
+  }
+
+  // Method to focus on the menu when expanded
+  public focus(): void {
+    if (this.expanded && this._menu) {
+      const menu = this._menu as MenuList;
+      Updates.enqueue(() => menu.focus());
+    }
+  }
+
+  // Method to toggle the expanded property
+  protected toggleExpanded = (e: MouseEvent) => {
+    this.expanded = !this.expanded;
+  };
+
+  // Method to close the menu
+  private closeMenu() {
+    this.expanded = false;
+  }
+
+  // Method to handle document click events
+  private handleDocumentClick = (e: any) => {
+    if (e && !e.composedPath().includes(this._menu) && !e.composedPath().includes(this._trigger)) {
+      this.closeMenu();
+    }
+  };
+
+  // Method to cleanup trigger event listeners
+  private cleanupTriggerEventListeners(trigger: HTMLElement): void {
+    if (this._trigger) {
+      if (this.openOnHover) {
+        this._trigger.removeEventListener('mouseover', this.toggleExpanded);
+      } else {
+        this._trigger.removeEventListener('click', this.toggleExpanded);
+      }
+    }
+  }
+  /**
+   * Handle keyboard interaction for the menu.
+   *
+   * @param e - the keyboard event
+   * @internal
+   */
+  public keydownHandler(e: KeyboardEvent): boolean | void {
+    const key = e.key || e.key.charCodeAt(0);
+
+    switch (key) {
+      case keyTab: {
+        if (this.expanded) {
+          this.expanded = false;
+        }
+        break;
+      }
+      case keyEnter: {
+        e.preventDefault();
+        this.expanded = !this.expanded;
+        break;
+      }
+
+      case keyEscape: {
+        if (this.expanded) {
+          e.preventDefault();
+          this.expanded = false;
+        }
+        break;
+      }
+      case keyHome:
+      case keyEnd: {
+        e.preventDefault();
+        break;
+      }
+      case keySpace: {
+        e.preventDefault();
+        this.expanded = !this.expanded;
+        break;
+      }
```

Comment:
I don't think these should be needed - is there a reason they're included? Space has the same problems as enter above.